DataOnion/Auth/SlidingExpirationStrategy.cs

```diff
@@ -0,0 +1,105 @@
+using StackExchange.Redis;
+
+namespace DataOnion.Auth
+{
+    public class SlidingExpirationStrategy<T> : IAuthServiceStrategy<T>
+        where T : class, IAuthStorable<T>
+    {
+        public static string ExpiryKey { get; } = "expiration";
+
+        private readonly IDatabase _database;
+        private readonly TimeSpan _timeout;
+        private readonly Func<HashEntry[], T> _makeFromHash;
+
+        private string _expirationTimeStr => DateTimeOffset.UtcNow
+            .Add(_timeout)
+            .ToUnixTimeSeconds()
+            .ToString();
+
+        public SlidingExpirationStrategy(
+            IDatabase database,
+            TimeSpan timeout,
+            Func<HashEntry[], T> makeFromHash
```

Comment:
i think this could be made simpler by just passing the config object to the constructor

Reply:
Sorry, I'm not sure what you mean by this. What object(s) do you want me to pass the constructor?